### PR TITLE
Improve styling of Input and Select controls

### PR DESCRIPTION
**Pull request by nwmac**
This PR updates the styling of the Input and Select controls and fixes a number of rendering issues with them (and a couple of related controls that use them).

- Remove the background colour from these controls when active - we uses a greyish background which is easily confused with the control presentation when disabled.
- Consistently use the border colour for :focus and :hover states - we use a mix of box shadow and outline (which both paint outside of the bounding box)
- Usea border colour change on control :hover
- Use a border colour change on control :focus and highlight the label (when labelled) with the outline color
- Introduce a compact mode - labelled controls will now correctly size themselves depending on whether a label is provided (this can be overridden using the compact property) - this should make it easier to ensure controls have the height when arranged in columnar layouts
- Improve the presentation of the LabelWithSelect control so that the select part no longer looks disabled and so that it has the same :hover and :focus behaviour as a normal Select control
- Fix minor positioning issues with tooltip icons and dropdown arrows
- Change the style of select boxes to have a border in the outline colour that outlines the complete control, including the select, i.e.
Screenshot 2022-01-13 at 14 25 07
Note: This requires various css to ensure the corner borders are shown correctly when the dropdown drops up/down and when used as part of a LabelWithSelect control

- Improve the presentation of the select control when displaying tags:
Screenshot 2022-01-13 at 14 14 27

- Tweaks to the KeyValue, Taints and Tolerations controls to improve the presentation
- Added example pages to the Storybook

### CODE REVIEW

File: utils/select.js [hunk 1]
```diff
@@ -22,3 +24,44 @@ export function onClickOption(option, e) {
     this.$refs['select-input'].closeSearchOptions();
   }
 }
+
+// This is a simpler positiionner for the dropdown for a select control
```
Comment (richard-cox) [inline utils/select.js:28]: typo

**Review by richard-cox — CHANGES_REQUESTED** [review]
We might want to slightly tone down (darken) the input field hover border in dark mode. Just to make it more subtle.

I spotted the following issues (not sure all are related to this PR)
- Deployment / Resources Tab --> `Add Toleration` button text isn't right
image
image
- Deployment / Scaling and Upgrade Policy --> Select unit height is incorrect
  image
- User Avatar Menu --> On click of menu item the left/right margin/padding is removed 
  image
- Deployment / General Tab --> deployment's with long name aren't elipsed in the NameNSDescription component
  image

**Comment by nwmac** [comment]
@richard-cox Updated with fixes - button colour will be done in another PR. I couldn't see the issue with the user dropdown - looked the same to me as before.

**Review by richard-cox — CHANGES_REQUESTED** [review]
Not sure if I originally missed this, but the content of the namespace filter in the header wiggles on hover. Other hover border effects like the list filter work fine.

Otherwise LGTM

Update: The commits need to be squashed too, that last merge brought in a lot of history

**Comment by nwmac** [comment]
@richard-cox I believe this is all updated and good to go - NamespaceFilter was fixed separately. I've rebased this PR with master.

**Review by richard-cox — APPROVED** [review]

**Pull request merged**